samza-core/src/main/java/org/apache/samza/clustermanager/ContainerManager.java

```diff
@@ -412,16 +429,23 @@ private void writeContainerPlacementResponseMessage(ContainerPlacementRequestMes
   }
 
   /**
-   * If there is an existing inflight request or container is pending a start, the container placement action shall wait
-   * until this in-flight action is complete
+   * These are specific scenarios in which a placement action should wait for existing action to complete before it is executed
+   * 1. If there is an placement request in progress on active container
+   * 2. If there is an placement request is progress on any of its standby container
+   * 3. If the container itself is pending a start
    *
    * @param requestMessage container placement request message
    * @return true if action should be taken right now, false if it needs to wait to be taken in future
    */
   private boolean deQueueAction(ContainerPlacementRequestMessage requestMessage) {
     // Do not dequeue action wait for the in-flight action to complete
-    if (hasActiveContainerPlacementAction(requestMessage.getProcessorId())) {
-      LOG.info("ContainerPlacement request: {} is en-queued because container has an in-progress placement action", requestMessage);
+    if (hasActiveContainerPlacementAction(requestMessage.getProcessorId())
+        || checkStandbyOrActiveContainerHasActivePlacementAction(requestMessage.getProcessorId())) {
+      if (standbyContainerManager.isPresent()) {
```

Comment:
This logic seems twisted, 
we check if there there is an active-action on the container-id, or if there is an active-action on its active/standby counterparts, 
after that we check if there is a standby-container-manager present?

Would it be possible to 
a. first check if a standby-container-manager is present?
Or
b. can hasActiveContainerPlacementAction encapsulate the logic of checking a and checking with if there is an active-action on its active/standby counterparts of the given processor-id.

Reply:
I see your point, let me refactor it